lib/unparser/constants.rb

```diff
@@ -27,6 +27,70 @@ def self.symbol_set(enumerable)
       ** %
     )
 
+    OPERATOR_PRECEDENCE = {
```

Comment:
I wounder if instead to manually "count" the weight fixnums we can do something like this:

```ruby
OPERATOR_PRECEDENCE = [
  %i[and or],
  %i[not],
  %i[defined],
  # ...
].each_with_index.map { |types, weight| types.map { |type| [type, weight] } }.to_h
```

This removes the mental overhead to manually increase the weight each level. And is less error prone on edits.